components/homme/src/test_src/dcmip12_wrapper.F90

```diff
@@ -592,21 +587,14 @@ subroutine dcmip2012_test4_init(elem,hybrid,hvcoord,nets,nete)
     do k=1,nlevp
       pressure=hvcoord%hyai(k)*hvcoord%ps0 + hvcoord%hybi(k)*ps_test
       do j=1,np; do i=1,np
-        !call get_coordinates(lat,lon,hyai,hybi, i,j,k,elem(ie),hvcoord)
-        lon  = elem(ie)%spherep(i,j)%lon
-        lat  = elem(ie)%spherep(i,j)%lat
+        call get_coordinates(lat,lon,hyai,hybi, i,j,k,elem(ie),hvcoord)
 
         !test4_baroclinic_wave(moist,X,lon,lat,p,z,zcoords,u,v,w,t,phis,ps,rho,q,q1,q2)
         !moist 0 or 1, X is Earth scale factor, zcoord=0, q is vapor, q1, q2
         call test4_baroclinic_wave(dcmip4_moist,dcmip4_X,lon,lat,&
                                    pressure,z,zcoords,u,v,w,T,phis,ps,rho,q,q1,q2)
         qarray(1)=q; qarray(2)=q1; qarray(3)=q2;
-        !dp = pressure_thickness(ps,k,hvcoord)
-        if (k <= nlev) then
-          dp = (hvcoord%hyai(k+1)-hvcoord%hyai(k))*p0 + (hvcoord%hybi(k+1)-hvcoord%hybi(k))*ps
-        else
-          dp = (hvcoord%hyai(k)-hvcoord%hyai(k-1))*p0 + (hvcoord%hybi(k)-hvcoord%hybi(k-1))*ps
-        endif
+        dp = pressure_thickness(ps,k,hvcoord)
```

Comment:
When `k=nlevp`, this line will access `hyai` and `hybi` at index `nlevp+1` inside of `pressure_thickness`, which does not exist.  Perhaps a more concise change here would be to use `dp = pressure_thickness(ps,max(k,nlev),hvcoord)`